lib/widgets/sprite_button.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/widgets.dart';
+import '../sprite.dart';
+import 'package:meta/meta.dart';
+
+class SpriteButton extends StatefulWidget {
+  final VoidCallback onPressed;
+  final Widget label;
+  final Sprite sprite;
+  final Sprite pressedSprite;
+  final double width;
+  final double height;
+
+  SpriteButton({
+    @required this.onPressed,
+    @required this.label,
+    @required this.sprite,
+    @required this.pressedSprite,
+    this.width,
+    this.height,
+  });
+
+  @override
+  State createState() => _ButtonState();
+}
+
+class _ButtonState extends State<SpriteButton> {
+  bool _pressed = false;
+
+  @override
+  Widget build(_) {
+    final width = widget.width ?? 200;
+    final height = widget.height ?? 50;
```

Comment:
I rather have the default value be set on the SpriteButton class, what do you think? so that when you do spriteButton.width it's not null and this here becomes uneeded.

Reply:
done